### PR TITLE
Doc update for core:list class

### PR DESCRIPTION
**Pull request by jpsiitonen**
Updating example code blocks for core:List class and landing page

### CODE REVIEW

File: sdk/lib/core/list.dart [hunk 1]
```diff
@@ -23,17 +26,38 @@ part of dart.core;
...
-/// fixedLengthList[0] = 87;
-/// var growableList = [1, 2];
-/// growableList.length = 0;
-/// growableList.add(499);
-/// growableList[0] = 87;
+/// ```
+///
+/// Example of growable list:
+/// ```dart
+/// final growableList = []; // []
```
Comment (alexlindroos) [inline sdk/lib/core/list.dart]: Maybe add `print(growableList); // []` instead

File: sdk/lib/core/list.dart [hunk 2]
```diff
@@ -508,22 +681,32 @@ abstract class List<E> implements EfficientLengthIterable<E> {
...
+  /// ```
   E removeAt(int index);
 
   /// Removes and returns the last object in this list.
   ///
   /// The list must be growable and non-empty.
+  /// ```dart
+  /// final parts = ['head', 'shoulder', 'knees', 'toes'];
+  /// final retVal = parts.removeLast(); // toes
+  /// print(parts); // [head, shoulder, toes]
```
Comment (alexlindroos) [inline sdk/lib/core/list.dart]: [head, shoulder, knees] is the right print

File: sdk/lib/core/list.dart [hunk 3]
```diff
@@ -556,14 +739,17 @@ abstract class List<E> implements EfficientLengthIterable<E> {
...
+  /// const end = 3;
+  /// print(colors.sublist(start, end)); // [green, blue]
   /// ```
   ///
   /// If [end] is omitted, it defaults to the [length] of this list.
   ///
   /// ```dart
-  /// print(colors.sublist(1)); // [green, blue, orange, pink]
+  /// final colors = ['red', 'green', 'blue', 'orange', 'pink'];
+  /// print(colors.sublist(3)); // [green, blue, orange, pink]
```
Comment (alexlindroos) [inline sdk/lib/core/list.dart]: Should the print be `[orange, pink]`

File: sdk/lib/core/list.dart (hunk 1 above)
Comment (jpsiitonen) [reply sdk/lib/core/list.dart]: done.

File: sdk/lib/core/list.dart [hunk 4]
```diff
@@ -23,17 +26,39 @@ part of dart.core;
...
+/// growableList[0] = 'G';
+/// print(growableList); // [G, B, C, D]
+/// // Concatenate items to single string
+/// print(growableList.join('-')); // G-B-C-D
+/// // Check is item on list
+/// final isA = growableList.contains('A'); // false
+/// // Sort items on list and iterate list content
+/// growableList.sort((a, b) => a.compareTo(b));
+/// print(growableList); // [B, C, D, G]
+/// // Check is value 'E' on list otherwise return 'N'
```
Comment (alexlindroos) [inline sdk/lib/core/list.dart]: nit: add line breaks to this chunk of code to make it more readable

**Review by alexlindroos — CHANGES_REQUESTED** [review]

File: sdk/lib/core/list.dart (hunk 3 above)
Comment (jpsiitonen) [reply sdk/lib/core/list.dart]: yes

**Review by alexlindroos — APPROVED** [review]

**Review by stepthomp — APPROVED** [review]

**Pull request closed without merging**